src/cls/rbd/cls_rbd.cc

```diff
@@ -6898,17 +6898,71 @@ int dir_remove(cls_method_context_t hctx,
 }
 
 static const string RBD_GROUP_SNAP_KEY_PREFIX = "snapshot_";
+static const string RBD_GROUP_SNAP_ORDER_KEY_PREFIX = "snap_order_";
+static const string RBD_GROUP_SNAP_MAX_ORDER_KEY = "snap_max_order";
 
 std::string snap_key(const std::string &snap_id) {
   ostringstream oss;
   oss << RBD_GROUP_SNAP_KEY_PREFIX << snap_id;
   return oss.str();
 }
 
+std::string snap_order_key(const std::string &snap_id) {
+  ostringstream oss;
+  oss << RBD_GROUP_SNAP_ORDER_KEY_PREFIX << snap_id;
+  return oss.str();
+}
+
+static std::string snap_id_from_order_key(const string &key) {
+  return key.substr(RBD_GROUP_SNAP_ORDER_KEY_PREFIX.size());
+}
+
+int snap_list_order(cls_method_context_t hctx,
```

Comment:
It doesn't look like this needs to be abstracted into a helper.  It makes sense for `snap_list()` which is used from two places (the other caller is `check_duplicate_snap_name()`), but not for `snap_list_order()`.

Let's move the body to `group_snap_list_order()` -- this would avoid the call to `clear()` as the map is actually created there and make it similar to e.g. `group_image_list()`.